mllib/src/main/scala/org/apache/spark/ml/tree/treeParams.scala

```diff
@@ -214,7 +216,7 @@ private[ml] object TreeClassifierParams {
 /**
  * Parameters for Decision Tree-based regression algorithms.
  */
-private[ml] trait TreeRegressorParams extends Params {
+private[ml] trait TreeRegressorParams extends DecisionTreeParams with HasVarianceCol {
```

Comment:
But if we only add ```HasVarianceCol``` to ```DecisionTreeRegressor```, we will need to implement ```validateAndTransformSchema``` for both ```DecisionTreeRegressor``` and ```DecisionTreeRegressionModel```. And further more, if we want to add ```HasVarianceCol``` to other Regressors, we need to implement ```validateAndTransformSchema``` for each *Regressor and *RegressionModel which is redundancy. I referred AFTSurvivalRegression which provide a similar prediction function ```predictQuantiles```. Looking forward to your comments.